tools/third_party_modified/mozlog/tests/test_structured.py

```diff
@@ -0,0 +1,1140 @@
+        self.logger.suite_end()
+
+
+class TestComponentFilter(BaseStructuredTest):
+    def test_filter_component(self):
+        component_logger = structuredlog.StructuredLogger(
+            self.logger.name, "test_component"
+        )
+        component_logger.component_filter = handlers.LogLevelFilter(lambda x: x, "info")
+
+        self.logger.debug("Test")
+        self.assertFalse(self.handler.empty)
+        self.assert_log_equals({"action": "log", "level": "DEBUG", "message": "Test"})
+        self.assertTrue(self.handler.empty)
+
+        component_logger.info("Test 1")
+        self.assertFalse(self.handler.empty)
+        self.assert_log_equals(
+            {
+                "action": "log",
+                "level": "INFO",
+                "message": "Test 1",
+                "component": "test_component",
+            }
+        )
+
+        component_logger.debug("Test 2")
+        self.assertTrue(self.handler.empty)
+
+        component_logger.component_filter = None
+
+        component_logger.debug("Test 3")
+        self.assertFalse(self.handler.empty)
+        self.assert_log_equals(
+            {
+                "action": "log",
+                "level": "DEBUG",
+                "message": "Test 3",
+                "component": "test_component",
+            }
+        )
+
+    def test_filter_default_component(self):
+        component_logger = structuredlog.StructuredLogger(
+            self.logger.name, "test_component"
+        )
+
+        self.logger.debug("Test")
+        self.assertFalse(self.handler.empty)
+        self.assert_log_equals({"action": "log", "level": "DEBUG", "message": "Test"})
+
+        self.logger.component_filter = handlers.LogLevelFilter(lambda x: x, "info")
+
+        self.logger.debug("Test 1")
+        self.assertTrue(self.handler.empty)
+
+        component_logger.debug("Test 2")
+        self.assertFalse(self.handler.empty)
+        self.assert_log_equals(
+            {
+                "action": "log",
+                "level": "DEBUG",
+                "message": "Test 2",
+                "component": "test_component",
+            }
+        )
+
+        self.logger.component_filter = None
+
+        self.logger.debug("Test 3")
+        self.assertFalse(self.handler.empty)
+        self.assert_log_equals({"action": "log", "level": "DEBUG", "message": "Test 3"})
+
+    def test_filter_message_mutuate(self):
+        def filter_mutate(msg):
+            if msg["action"] == "log":
+                msg["message"] = "FILTERED! %s" % msg["message"]
+            return msg
+
+        self.logger.component_filter = filter_mutate
+        self.logger.debug("Test")
+        self.assert_log_equals(
+            {"action": "log", "level": "DEBUG", "message": "FILTERED! Test"}
+        )
+        self.logger.component_filter = None
+
+
+class TestCommandline(unittest.TestCase):
+    def get_logfile(self):
+        return tempfile.NamedTemporaryFile()
```

Comment:
Looks like Windows [cannot open two handles to the same temporary file](https://stackoverflow.com/questions/23212435/permission-denied-to-write-to-my-temporary-file) simultaneously, so keeping this `tempfile.NamedTemporaryFile()` open during the test will conflict with [`open(...)`](https://github.com/web-platform-tests/wpt/pull/41183/files#diff-ceb6ab4ddc17073946cc8c6d8d94351b987f73436b0c30d549312f7976f6f9afR136)  from one of the parser's `type=...` handlers. I would suggest closing the file after it's allocated, since the tests currently only access the file's `.name`.

```suggestion
    @contextlib.contextmanager
    def get_logfile(self):
        tmp_file = tempfile.NamedTemporaryFile(delete=False)
        tmp_file.close()
        try:
            yield tmp_file
        finally:
            os.unlink(tmp_file.name)
```

Then, pass the file opened by the parser to `loglines(...)` instead, like so:

```
args = parser.parse_args(['--log-tbpl=...'])
...
self.assertEqual([...], self.loglines(args.log_tbpl))
```

(If there are problems `unlink()`ing the file with an open handle, we might need to close the `args`-side handle first.)